File: game/forest/hurt_branch_talking.py
import pygame

from animation import Animation
from forest.forest_resources import ForestResources
from forest.forest_utils import ForestUtils
from render_type import RenderType
from state import State


class HurtBranchTalking(State):
    def __init__(self, context):
        super().__init__(context)
        self.needs_bottom = RenderType.POST

    def process_events(self, events):
        if self.get_frame_index() >= len(ForestResources.sync_hitlog):
            return ForestUtils.reduce_lives(self.context)
        return None

    def render(self, screen):
        screen.blit(Animation.get_sync_frame(ForestResources.hugohitlog_talk, ForestResources.sync_hitlog, self.get_frame_index()), (0, 0))

    def on_enter(self) -> None:
        super().on_enter()
        pygame.mixer.Sound.play(ForestResources.speak_hitlog)

File: game-web/src/forest/hurtBranchTalking.ts
/** Port of game/forest/hurt_branch_talking.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { ForestResources } from './forestResources';
import { ForestUtils } from './forestUtils';
import { RenderType } from '../renderType';
import type { GameData } from '../gameData';
import type { PhoneEvents } from '../phoneEvents';

export class HurtBranchTalking extends State {
  readonly needsBottom = RenderType.POST;

  constructor(context: GameData) { super(context); }

  processEvents(_events: PhoneEvents): StateClass | null {
    if (this.getFrameIndex() >= ForestResources.syncHitlog.length) {
      return ForestUtils.reduceLives(this.context, 4); // Branch
    }
    return null;
  }

  onEnter(): void {
    super.onEnter();
    AudioHelper.play(ForestResources.speakHitlog);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getSyncFrame(ForestResources.hugohitlogTalk, ForestResources.syncHitlog, this.getFrameIndex()), 0, 0);
  }
}
